apps/prairielearn/src/pages/home/home.tsx

```diff
@@ -121,15 +128,45 @@ router.post(
     } = getPageContext(res.locals, { withAuthzData: false });
 
     if (body.__action === 'accept_invitation') {
-      await execute(sql.accept_invitation, {
+      await ensureEnrollment({
         course_instance_id: body.course_instance_id,
-        uid,
         user_id,
+        agent_user_id: user_id,
+        agent_authn_user_id: user_id,
+        action_detail: 'invitation_accepted',
       });
     } else if (body.__action === 'reject_invitation') {
-      await execute(sql.reject_invitation, {
-        course_instance_id: body.course_instance_id,
-        uid,
+      await runInTransactionAsync(async () => {
```

Comment:
Conventionally this would all be a model function, though we can save that refactor for later.